spec/lib/vanagon/driver_spec.rb

```diff
@@ -54,6 +54,40 @@ def create_driver(platform, options = {})
                               'engine' => 'hardware' })
     end
 
+    it 'returns the platform and a scheduler engine when scheduled resources are enabled and an engine is specified' do
+      ENV['VANAGON_SCHEDULED_HARDWARE'] = "true"
+
+      platform = eval_platform('aix-7.1-ppc', <<-END)
+        platform 'aix-7.1-ppc' do |plat|
+          plat.build_host ["pe-aix-71-01", "pe-aix-71-02"]
+          plat.engine "scheduled"
+        end
+      END
+
+      info = create_driver(platform).build_host_info
+
+      expect(info).to match({ 'name'   => 'aix-7.1-ppc',
```

Comment:
I think this needs more discussion.  I have a candidate implementation for this up at @320891bd2b0.  Some thoughts (some mentioned in the commit message):

This is an explicit translation of known hardware platforms in use here (Puppet). An explicit translation list has a number of disadvantages:

 - When we add platforms we have to update it
 - It is necessarily focused on our conventions and installations
 - Which means it makes life hard for other vanagon users
 - It doesn't vary with time or versions of related software (so we can have difficulties with drift, or with trying to move from one implementation to another)

It would be possible to make the translation more programmatic (like via a regex), but that has most of the same pitfalls, even if we don't have to edit the file as often.

We've not been using hardware resources in ABS as of yet. We could translate on the inbound side if we wanted to (though for AIX, the vmpooler "style" is never going to actually touch vmpooler, so it's completely a "style").  We could not translate at all and just pass the platform through, even though it's not in the vmpooler "style".

We could bite the bullet (though I never want to make it be a dependency of much of anything) and finally implement a library that understands how to translate between platform standards.

Anyway, I think more discussion around this is needed. It's easy to revert a commit, so no worries there.